app/tests/spec/lib/country-telephone-info.js

```diff
@@ -90,6 +171,48 @@ define((require, exports, module) => {
       });
     });
 
+    describe('LU', () => {
+      const { format, normalize, pattern } = CountryTelephoneInfo.LU;
+
+      describe('format', () => {
+        it('formats correctly', () => {
+          assert.equal(format('621123456'), '+352 621123456');
+        });
+      });
+
+      describe('normalize', () => {
+        it('normalizes a number accepted by pattern correctly', () => {
+          assert.equal(normalize('+352621123456'), '+352621123456');
+          assert.equal(normalize('621123456'), '+352621123456');
+        });
+      });
+
+      describe('pattern', () => {
+        it('validates correctly', () => {
+          // Incorrect mobile prefix (first character)
+          assert.isFalse(pattern.test('521123456'));
+          assert.isFalse(pattern.test('+352521123456'));
+
+          // Incorrect mobile prefix (third character)
+          assert.isFalse(pattern.test('622123456'));
+          assert.isFalse(pattern.test('+352622123456'));
+
+          // too short
+          assert.isFalse(pattern.test('62112345'));
+          assert.isFalse(pattern.test('+35262112345'));
+
+          // too long
+          assert.isFalse(pattern.test('6211234567'));
+          assert.isFalse(pattern.test('+3526211234567'));
+
+          assert.ok(pattern.test('621123456'));
+          assert.ok(pattern.test('+352621123456'));
+          assert.ok(pattern.test('631123456'));
+          assert.ok(pattern.test('+352631123456'));
+        });
+      });
+    });
+
```

Comment:
I just realised how big this test module is going to get when we add loads more countries.

Is there any mileage in defining the countries and their respective good/bad patterns as a blob of JSON, then doing a `forEach` over it with the assertions in there? Feel free to ignore if it's a rubbish suggestion.

Reply:
>  Feel free to ignore if it's a rubbish suggestion.

I like this suggestion.